### PR TITLE
Improved error message for multiple SNP variants found in fetch_snp

### PR DESCRIPTION
**Pull request by ChrOertlin**
This PR adds/fixes ...

This PR aims to improve an error message log in _fetch_snp()_ of the VCF.py file.

**How to prepare for test**:
- [x] ssh to Hasta
- [x] install on stage of Hasta:
`bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_genotype -t genotype -b improveErrorMsg`

**How to test**:
- [x] Login to the stage environment. Run "cg upload genotypes safemoose" as this case raises the associated error.     

**Expected test outcome**:
- [x] Check that  the message is understandable. It should list the number of variants found, the position where it is at, the actual variants and the associated bcf file.
- [x] Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] Tests executed by KSV & CO 
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

Implementation plan:
- [ ] Deploy to prod when tested

### CODE REVIEW

**Review by henrikstranneheim — APPROVED** [review]
Much clearer, 👍 !

**Comment by ChrOertlin** [comment]
@henrikstranneheim , could you have a second look at the code. To improve the formatting in the error message we had to write up a function that parses the Variant.

File: genotype/load/vcf.py [hunk 1]
```diff
@@ -64,4 +64,13 @@ def fetch_snp(vcf: VCF, snp: SNP) -> Variant:
         variant = variants[0]
         return variant
     # weird SNP position lookup, not even possible; right?
-    raise ValueError("multiple variants found for SNP")
+    raise ValueError(
+        f"Multiple variants ({len(variants)}) found for SNP at position {pos_str}: {return_alternatives(variants)}."
+    )
+
+
+def return_alternatives(variants: List[Variant]) -> str:
```
Comment (henrikstranneheim) [inline genotype/load/vcf.py:72]:
```suggestion
def return_multiple_variants_alleles(variants: List[Variant]) -> str:
```
Also always add a doc string

File: genotype/load/vcf.py [hunk 2]
```diff
@@ -64,4 +64,13 @@ def fetch_snp(vcf: VCF, snp: SNP) -> Variant:
...
+    alternatives = []
```
Comment (henrikstranneheim) [inline genotype/load/vcf.py:73]: Add type hint and perhaps rename the variable to something with allele to make it more clear what it represents

File: genotype/load/vcf.py [hunk 3]
```diff
@@ -64,4 +64,13 @@ def fetch_snp(vcf: VCF, snp: SNP) -> Variant:
         variant = variants[0]
         return variant
     # weird SNP position lookup, not even possible; right?
-    raise ValueError("multiple variants found for SNP")
+    raise ValueError(
+        f"Multiple variants ({len(variants)}) found for SNP at position {pos_str}: {return_alternatives(variants)}."
```
Comment (henrikstranneheim) [inline genotype/load/vcf.py:68]: Use named args whenever possible. It makes it more clear on what to supply to the function

**Review by henrikstranneheim — COMMENTED** [review]
Great that you are fixing this

File: genotype/load/vcf.py [hunk 4]
```diff
@@ -56,12 +56,21 @@ def fetch_snp(vcf: VCF, snp: SNP) -> Variant:
...
         return variant
     # weird SNP position lookup, not even possible; right?
-    raise ValueError("multiple variants found for SNP")
+    raise ValueError(
+        f"Multiple variants ({len(variants)}) found for SNP at position {pos_str}: {return_multiple_variant_alleles(variants=variants)}."
+    )
+
+
+def return_multiple_variant_alleles(variants: List[Variant]) -> str:
+    multiple_variant_alleles: List = []
```
Comment (henrikstranneheim) [inline genotype/load/vcf.py:73]: Just a doc string and we are good to go

**Review by henrikstranneheim — APPROVED** [review]
👍

**Comment by Karl-Svard** [comment]
## Deployed in prod
```
11d [hiseq.clinical@hasta:~] [P_base] $ bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-prod.sh -e P_genotype -t genotype -b master
Collecting git+https://github.com/Clinical-Genomics/genotype@master
  Running command git clone --quiet https://github.com/Clinical-Genomics/genotype /home/hiseq.clinical/tmp/pip-req-build-0nuksv2r
  Cloning https://github.com/Clinical-Genomics/genotype (to revision master) to ./tmp/pip-req-build-0nuksv2r
  Resolved https://github.com/Clinical-Genomics/genotype to commit b1df4c5468dacf051f66f57606bbeff867d91d8d
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Building wheels for collected packages: genotype
  Building wheel for genotype (pyproject.toml): started
  Building wheel for genotype (pyproject.toml): finished with status 'done'
  Created wheel for genotype: filename=genotype-4.1.4-py3-none-any.whl size=31431 sha256=4792d83d38455a0919fa3ab0ccde59708dac498f37571558b0191870e32886f1
  Stored in directory: /home/hiseq.clinical/tmp/pip-ephem-wheel-cache-ijsvo1nt/wheels/01/16/ea/57ccaf0014013064dba677747cfca144721c057caa9bbc6734
Successfully built genotype
Installing collected packages: genotype
  Attempting uninstall: genotype
    Found existing installation: genotype 4.1.2
    Uninstalling genotype-4.1.2:
      Successfully uninstalled genotype-4.1.2
Successfully installed genotype-4.1.4
Collecting git+https://github.com/Clinical-Genomics/genotype@master
  Running command git clone --quiet https://github.com/Clinical-Genomics/genotype /home/hiseq.clinical/tmp/pip-req-build-fh2xame0
  Cloning https://github.com/Clinical-Genomics/genotype (to revision master) to ./tmp/pip-req-build-fh2xame0
  Resolved https://github.com/Clinical-Genomics/genotype to commit b1df4c5468dacf051f66f57606bbeff867d91d8d
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Requirement already satisfied: pymysql in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.4) (1.0.2)
Requirement already satisfied: Flask==1.1.1 in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.4) (1.1.1)
Requirement already satisfied: gunicorn in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.4) (20.1.0)
Requirement already satisfied: markupsafe==1.1.1 in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.4) (1.1.1)
Requirement already satisfied: Flask-Bootstrap in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.4) (3.3.7.1)
Requirement already satisfied: cyvcf2 in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.4) (0.30.18)
Requirement already satisfied: Alchy in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.4) (2.2.2)
Requirement already satisfied: wheel in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.4) (0.37.1)
Requirement already satisfied: pysam in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.4) (0.19.1)
Requirement already satisfied: itsdangerous==1.1.0 in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.4) (1.1.0)
Requirement already satisfied: Flask-Login==0.5.0 in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.4) (0.5.0)
Requirement already satisfied: Flask-Reverse-Proxy in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.4) (0.2.0.2)
Requirement already satisfied: numpy==1.19.2 in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.4) (1.19.2)
Requirement already satisfied: ruamel.yaml in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.4) (0.17.21)
Requirement already satisfied: Flask-Security in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.4) (3.0.0)
Requirement already satisfied: Flask-Dance==3.0.0 in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.4) (3.0.0)
Requirement already satisfied: xlrd==1.2.0 in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.4) (1.2.0)
Requirement already satisfied: Flask-SQLAlchemy==2.1 in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.4) (2.1)
Requirement already satisfied: pyyaml in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.4) (6.0)
Requirement already satisfied: jinja2==2.11.2 in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.4) (2.11.2)
Requirement already satisfied: click in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.4) (8.1.3)
Requirement already satisfied: coloredlogs in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.4) (15.0.1)
Requirement already satisfied: Flask-Alchy in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.4) (0.5.0)
Requirement already satisfied: cython in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.4) (0.29.32)
Requirement already satisfied: werkzeug<1.0.0 in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.4) (0.16.1)
Requirement already satisfied: SQLAlchemy<1.4 in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.4) (1.3.24)
Requirement already satisfied: oauthlib in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from Flask-Dance==3.0.0->genotype==4.1.4) (3.2.2)
Requirement already satisfied: six in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from Flask-Dance==3.0.0->genotype==4.1.4) (1.16.0)
Requirement already satisfied: urlobject in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from Flask-Dance==3.0.0->genotype==4.1.4) (2.4.3)
Requirement already satisfied: requests-oauthlib>=1.0.0 in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from Flask-Dance==3.0.0->genotype==4.1.4) (1.3.1)
Requirement already satisfied: requests>=2.0 in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from Flask-Dance==3.0.0->genotype==4.1.4) (2.28.1)
Requirement already satisfied: importlib-metadata in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from click->genotype==4.1.4) (5.0.0)
Requirement already satisfied: pydash>=3.4.7 in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from Alchy->genotype==4.1.4) (5.1.1)
Requirement already satisfied: humanfriendly>=9.1 in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from coloredlogs->genotype==4.1.4) (10.0)
Requirement already satisfied: visitor in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from Flask-Bootstrap->genotype==4.1.4) (0.1.3)
Requirement already satisfied: dominate in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from Flask-Bootstrap->genotype==4.1.4) (2.7.0)
Requirement already satisfied: Flask-BabelEx>=0.9.3 in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from Flask-Security->genotype==4.1.4) (0.9.4)
Requirement already satisfied: Flask-WTF>=0.13.1 in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from Flask-Security->genotype==4.1.4) (1.0.1)
Requirement already satisfied: Flask-Principal>=0.3.3 in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from Flask-Security->genotype==4.1.4) (0.4.0)
Requirement already satisfied: Flask-Mail>=0.7.3 in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from Flask-Security->genotype==4.1.4) (0.9.1)
Requirement already satisfied: passlib>=1.7 in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from Flask-Security->genotype==4.1.4) (1.7.4)
Requirement already satisfied: setuptools>=3.0 in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from gunicorn->genotype==4.1.4) (65.5.0)
Requirement already satisfied: ruamel.yaml.clib>=0.2.6 in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from ruamel.yaml->genotype==4.1.4) (0.2.7)
Requirement already satisfied: Babel>=1.0 in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from Flask-BabelEx>=0.9.3->Flask-Security->genotype==4.1.4) (2.10.3)
Requirement already satisfied: speaklater>=1.2 in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from Flask-BabelEx>=0.9.3->Flask-Security->genotype==4.1.4) (1.3)
Requirement already satisfied: blinker in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from Flask-Mail>=0.7.3->Flask-Security->genotype==4.1.4) (1.5)
Requirement already satisfied: WTForms in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from Flask-WTF>=0.13.1->Flask-Security->genotype==4.1.4) (3.0.1)
Requirement already satisfied: certifi>=2017.4.17 in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from requests>=2.0->Flask-Dance==3.0.0->genotype==4.1.4) (2022.9.24)
Requirement already satisfied: idna<4,>=2.5 in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from requests>=2.0->Flask-Dance==3.0.0->genotype==4.1.4) (3.4)
Requirement already satisfied: charset-normalizer<3,>=2 in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from requests>=2.0->Flask-Dance==3.0.0->genotype==4.1.4) (2.1.1)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from requests>=2.0->Flask-Dance==3.0.0->genotype==4.1.4) (1.26.12)
Requirement already satisfied: zipp>=0.5 in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from importlib-metadata->click->genotype==4.1.4) (3.10.0)
Requirement already satisfied: typing-extensions>=3.6.4 in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from importlib-metadata->click->genotype==4.1.4) (4.4.0)
Requirement already satisfied: pytz>=2015.7 in /home/proj/production/bin/miniconda3/envs/P_genotype/lib/python3.7/site-packages (from Babel>=1.0->Flask-BabelEx>=0.9.3->Flask-Security->genotype==4.1.4) (2022.5)
fatal: Not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
repository is clean
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/hiseq.clinical
done.
Log deploy... done.
genotype, version 4.1.4
```

**Pull request merged**